vita3k/compat/src/compat.cpp

```diff
@@ -43,6 +44,42 @@ static std::string db_updated_at;
 static const uint32_t db_version = 1;
 static uint32_t db_issue_count = 0;
 
+bool extract_zip_file(const char *zip_filename, const fs::path &output_path) {
+    // Open the ZIP file for reading
+    mz_zip_archive zip_archive;
+    memset(&zip_archive, 0, sizeof(zip_archive));
+    if (!mz_zip_reader_init_file(&zip_archive, zip_filename, 0)) {
+        LOG_ERROR("Failed to initialize ZIP archive for reading");
+        return false;
+    }
+
+    // Get the number of files in the ZIP archive
+    mz_uint num_files = mz_zip_reader_get_num_files(&zip_archive);
+    for (mz_uint i = 0; i < num_files; i++) {
+        // Get information about the current file in the ZIP archive
+        mz_zip_archive_file_stat file_stat;
+        if (!mz_zip_reader_file_stat(&zip_archive, i, &file_stat)) {
+            LOG_ERROR("Failed to get file information from ZIP archive");
+            mz_zip_reader_end(&zip_archive);
+            return false;
+        }
+
+        // Extract the file from the ZIP archive
+        fs::path output_file_path = output_path / file_stat.m_filename;
+        if (!mz_zip_reader_extract_to_file(&zip_archive, i, output_file_path.string().c_str(), 0)) {
```

Comment:
`fs_utils::path_to_utf8` also needs to be used here

Reply:
Oups 🤦‍♂️